JS.md

```diff
@@ -0,0 +1,233 @@
+# JavaScript API
+
+In the [MVP](MVP.md), the only way to access WebAssembly on the Web is through
+an explicit JS API which is defined below. (In the future, WebAssembly may also
+be loaded and run directly from an HTML `<script type='module'>` tag—and
+any other Web API that loads ES6 modules via URL—as part of 
+[ES6 Module integration](Modules.md#integration-with-es6-modules).)
+
+## The `Wasm` object
+
+The `Wasm` object is the initial value of the `Wasm` property of the global
+object. Like the `Math` and `JSON` objects, the `Wasm` object is a plain
+JS object (not a constructor or function) that acts like a namespace and has the
+following properties:
+
+### Constructor Properties of the `Wasm` object
+
+* `Wasm.Module` : see [`Wasm.Module` objects](#wasmmodule-objects)
+* `Wasm.Instance` : see [`Wasm.Instance` objects](#wasminstance-objects)
+* `Wasm.CompileError` : a [NativeError](http://tc39.github.io/ecma262/#sec-nativeerror-object-structure)
+   which indicates an error during WebAssembly decoding or validation
+* `Wasm.RuntimeError` : a [NativeError](http://tc39.github.io/ecma262/#sec-nativeerror-object-structure)
+   which indicates an error while running WebAssembly code
+
+### Function Properties of the `Wasm` object
+
+#### `Wasm.compile`
+
+The `compile` function has the signature:
+```
+Promise<Wasm.Module> compile(TypedArray bytes)
+```
+If the given `bytes` argument is not a
+[TypedArray object](http://tc39.github.io/ecma262/#sec-typedarray-objects),
+a `TypeError` exception is thrown.
+
+Otherwise, this function returns a `Promise` for an asychronous task
+to compile a `Wasm.Module` as described in the 
+[`Wasm.Module` constructor](#wasmmodule-constructor). On success, the
+`Promise` is [fulfilled](http://tc39.github.io/ecma262/#sec-fulfillpromise)
+with the resulting `Wasm.Module` instance. On failure, the `Promise` is 
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise) with the 
+`Wasm.CompileError`.
+
+The asynchronous compilation is performed on the state of the given
+TypedArray when `compile` is first called; subsequent mutations of the
+TypedArray does not affect ongoing compilations.
```

Comment:
Yes, but if we add the "opt-in to shared memory" attribute to WebIDL, then this shouldn't be an issue, right?